classes/Presenter/Order/OrderPendingPresenter.php

```diff
@@ -0,0 +1,114 @@
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author    PrestaShop SA <contact@prestashop.com>
+*  @copyright 2007-2019 PrestaShop SA
+*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+namespace PrestaShop\Module\PrestashopCheckout\Presenter\Order;
+
+use PrestaShop\Module\PrestashopCheckout\Adapter\LinkAdapter;
+use PrestaShop\Module\PrestashopCheckout\OrderStates;
+use PrestaShop\Module\PrestashopCheckout\Presenter\PresenterInterface;
+use PrestaShop\Module\PrestashopCheckout\Translations\OrderStatesTranslations;
+
+/**
+ * Present the pending orders for the reporting
+ */
+class OrderPendingPresenter implements PresenterInterface
+{
+    /**
+     * present pending orders
+     *
+     * @return array
+     */
+    public function present()
+    {
+        $link = new LinkAdapter();
+        $context = \Context::getContext();
+
+        $orderStates = [];
+        $orderTranslations = new OrderStatesTranslations();
+        $orderTranslations = $orderTranslations->getTranslations($context->language->iso_code);
+        foreach (OrderStates::ORDER_STATES as $key => $value) {
+            if ($key == 'PS_CHECKOUT_STATE_WAITING_PAYPAL_PAYMENT' ||
+                $key == 'PS_CHECKOUT_STATE_WAITING_CREDIT_CARD_PAYMENT' ||
+                $key == 'PS_CHECKOUT_STATE_WAITING_LOCAL_PAYMENT' ||
+                $key == 'PS_CHECKOUT_STATE_WAITING_CAPTURE'
+            ) {
+                $idState = \Configuration::get($key);
+                $orderStates[$idState]['color'] = $value;
+                $orderStates[$idState]['name'] = $orderTranslations[$key];
+            }
+        }
+
+        $orders = $this->getPendingOrders($orderStates);
+
+        foreach ($orders as &$order) {
+            $order['username'] = $this->getUsername($order['id_customer']);
+            $order['userProfileLink'] = $link->getAdminLink('AdminCustomers', true, [], ['id_customer' => $order['id_customer'], 'viewcustomer' => 1]);
+            $order['orderLink'] = $link->getAdminLink('AdminOrders', true, [], ['id_order' => $order['id_order'], 'vieworder' => 1]);
+            $order['state'] = $orderStates[$order['current_state']];
+            $order['before_commission'] = \Tools::displayPrice($order['total_paid']);
+            // TODO: Waiting for paypal infos (reporting lot 2)
+            $order['commission'] = '-';
+            $order['total_paid'] = '-';
+        }
+
+        return $orders;
+    }
+
+    /**
+     * getUsername
+     *
+     * @param string|int $userID
+     *
+     * @return string
+     */
+    private function getUsername($userID)
+    {
+        $sql = 'SELECT firstname,lastname FROM `' . _DB_PREFIX_ . 'customer` o WHERE id_customer = ' . $userID;
```

Comment:
I don't know if it's usefull but we also can do : 
`return \db::getInstance()->getValue('SELECT CONCAT(SUBSTR(firstname, 1, 1), ". ", lastname) FROM `' . _DB_PREFIX_ . 'customer` o WHERE id_customer = ' . $userID);`

Reply:
ye ok thx for the tip, not really better or something just different in our case